doc/spec_TransferElement.dox

```diff
@@ -342,16 +343,20 @@ This documnent is currently still **INCOMPLETE**!
+  - \anchor transferElement_D_1_1 1.1 The TransferElement::_readQueue is a cppext::future_queue of type 'void' which exists in each transfer element. For transfer elements with AccessMode::wait_for_new_data it usually is a deferred continuation queue of another future_queue which is transporting the data (\anchor transferElement_data_transport_queue called *implementation-specific data transport queue* throughout this document) \ref transferElement_comment_D_1_1 "(*)". The data type of this queue is implementation dependent. Common functionality like exception handling and waiting for new data is implemented on TransferElement::_readQueue.
   - 1.2 TransferElement::_readQueue is initialised in the constructor. [TBD: Allow setting the queue length through the public API?]
-  - 1.3 TransferElement::_readQueue is pushed whenever new data has arrived. If important for the implementation, the return value of cppext::future_queue::push_overwirte() will tell whether data has been discarded (*).
+  - 1.3 *implementation-specific data transport queue* is pushed whenever new data has arrived. If important for the implementation, the return value of cppext::future_queue::push_overwirte() will tell whether data has been discarded (*).
+  - \anchor transferElement_D_1_4 1.4 In case an exception is detected during an asychronous transfer (for instance in a separate thread), the exception must be pushed to the *implementation-specific data transport queue* so it can be received through the TransferElement::_readQueue \ref transferElement_comment_D_1_4 "(*)". The TransferElement implementation will then make sure that the exception is properly rethrown in postRead (just like for synchronous transfers).
 
-  - 1.4 In case an exception is detected during an asychronous transfer (for instance in a separate thread), the exception must be stored on the TransferElement::_readQueue. The TransferElement implementation will then make sure that the exception is properly rethrown in postRead (just like for synchronous transfers).
 - 2. In doPostRead() the whole application buffer (data buffer and meta data) must be updated together if updateDataBuffer is true and there was no exception, or not at all in all other cases.
 
 ### (*) Comments ###
 
-1.3 Either the currently pushed data or older data on the queue might be discarded. In any case there will be one call less to TransferElement::doPostRead(), because the number of entries in the queue could not be increased because it was full
+- \anchor transferElement_comment_D_1_1 \ref transferElement_D_1_1 "1.1" As the *implementation-specific data transport queue* has to work with push_overwrite_exception, it has to have a minimum length of 2.
```

Comment:
It also has to use push_overwrite for the non-exception data, which also results in the same minimum length requirement.